bin/ci-publish.sh

```diff
@@ -3,10 +3,14 @@ set -eu
 PUBLISH=${CI_PUBLISH:-false}
 SECURE_VAR=${TRAVIS_SECURE_ENV_VARS:-false}
 
-if [[ "$TRAVIS_SECURE_ENV_VARS" == true && -n "$TRAVIS_TAG" && "$CI_PUBLISH" == true ]]; then
+if [[ "$TRAVIS_SECURE_ENV_VARS" == true && "$CI_PUBLISH" == true ]]; then
   git log | head -n 20
   echo "$PGP_SECRET" | base64 --decode | gpg --import
-  sbt ci-publish
+  if [ -n "$TRAVIS_TAG" ]; then
+    sbt ci-publish
+  else
+    sbt -Dscalameta.snapshot=true ci-publish
+  fi
```

Comment:
Why not publish prereleases on every merge like we did previously?

Reply:
bintray is unstable, has file size limits, and requires different credentials and sbt plugins to work. This change is a tiny change from our current setup

Reply:
publishing non-SNAPSHOT on every merge to maven central creates noise

Reply:
What do you mean by noise? Are we risking to run into quota issues? Is there an important webpage that would be spammed by frequent releases?

Reply:
see https://github.com/scalameta/scalameta/pull/1225#issuecomment-359033250 publishing -SNAPSHOT to sonatype is logistically simpler than non-SNAPSHOT releases

Reply:
Okay, that's a good reason. Can you talk a bit more about the workflow?

1) So, the idea is that we publish -SNAPSHOT releases to Sonatype releases but don't promote them to Maven Central?

2) Can there be a possibility to run `sonatypeRelease` after all the CI jobs have finished? If there was such a possibility, would we still need -SNAPSHOT, or we would be able to switch to regular prereleases?

Reply:
1. Correct, we publish -SNAPSHOT on every merge and forget about them, they never appear in search results in http://search.maven.org/, only https://oss.sonatype.org/content/repositories/snapshots/org/scalameta/

2. I don't want every merge to go to maven central even if I could. I don't want them to appear on http://search.maven.org/  Stable  releases should be accompanied with a tag, changelist in GitHub releases and attributes to contributors. Stable releases should be something worth sharing on gitter/twitter.

Reply:
A stable release is still only a tag push away, I don't see any of my projects ending up depending on -SNAPSHOT releases for long periods. I intend to continue pushing tags liberally like I did this morning.

Reply:
> Can there be a possibility to run sonatypeRelease after all the CI jobs have finished? 

I think it can be done with Travis build stages, which is an experimental feature. We're using it in metals with good success.